packages/app-layout-element/src/browser/grey-element.mjs

```diff
@@ -0,0 +1,364 @@
+
+    let style = document.createElement('style')
+    style.textContent = STYLES
+    shadow.appendChild(style)
+    style = document.createElement('style')
+    style.setAttribute('media', 'print')
+    style.textContent = STYLES_PRINT
+    shadow.appendChild(style)
+
+    const template = document.createElement('template')
+    template.innerHTML = TEMPLATE
+    const elBody = template.content.cloneNode(true)
+    const elRoot = document.createElement('div')
+    elRoot.setAttribute('class', 'app-layout-body')
+    elRoot.dataset.componentLocalName = this.localName
+    elRoot.setAttribute('id', 'app-layout')
+    elRoot.appendChild(elBody)
+    shadow.appendChild(elRoot)
+
+    this.#onConstructorEnd()
+  }
+
+  async connectedCallback() {
+    if (!this.isConnected) return
+    await this.#mutateHostForStyleSheets()
+  }
+
+  #onConstructorEnd = () => {
+    this.#enforceOnlyChildOfParent()
+    this.#mutateHostHeadMetaViewport()
+  }
+
+  /**
+   * Enforce idea that this component must be as first child of body to ensure
+   * that this component takes up all the available space as designed
+   */
+  #enforceOnlyChildOfParent = () => {
+    const isBody = this.parentElement?.localName === 'body'
+    if (!isBody) {
+      const message = `Invalid location to use <${this.localName} />, make sure it's a direct descendant of body`
+      console.error(message)
+      if (this.shadowRoot == null) return
+      this.shadowRoot.innerHTML = `<div style="color: red;">${message}</div>`
+      throw new Error(message)
+    }
+  }
+
+  #mutateHostHeadMetaViewport = () => {
+    if (this.shadowRoot == null) return
+    const d = this.shadowRoot.ownerDocument
+    const metaViewport = d.querySelector('head > meta[name="viewport"]')
+    if (!metaViewport) {
+      const newMetaViewport = d.createElement('meta')
+      newMetaViewport.setAttribute('name', 'viewport')
+      newMetaViewport.setAttribute(
+        'content',
+        'width=device-width, initial-scale=1, viewport-fit=cover, user-scalable=no',
+      )
+      newMetaViewport.setAttribute('data-comment-for', this.localName)
+      d.head.appendChild(newMetaViewport)
+    }
+  }
+
+  #mutateHostForStyleSheets = async () => {
+    try {
+      const styles = await optimizedExternalStyles(
+        window,
+        LIST_EXTERNAL_STYLE,
+        {
+          componentName: this.localName,
+          avoidDuplicates: true,
+        },
+      )
+      // Apply to Shadow DOM
+      if (styles.shadow instanceof CSSStyleSheet) {
+        this.shadowRoot.adoptedStyleSheets = [styles.shadow]
+      } else if (styles.shadow) {
+        this.shadowRoot.appendChild(styles.shadow)
+      }
+
+      // Apply to document head if not already there
+      if (styles.document && !styles.document.isConnected) {
+        document.head.appendChild(styles.document)
+      }
+    } catch (error) {
+      console.error('Style loading failed:', error)
+    }
+  }
```

Comment:
This is more like copy-pasta. Probably that should be factored out similarly to what's described in #39